src/main/resources/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote/script.js

```diff
@@ -14,8 +14,14 @@ Behaviour.specify("span.pipeline-new-node", 'NewNodeConsoleNote', 0, function(e)
     var nodeId = e.getAttribute('nodeId')
     var startId = e.getAttribute('startId')
     if (startId == null || startId == nodeId) {
-        e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" onclick="showHidePipelineSection(this); return false">hide</a>)</span>')
+        e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" class="pipeline-toggle">hide</a>)</span>')
         // TODO automatically hide second and subsequent branches: namely, in case a node has the same parent as an earlier one
+        const toggles = e.querySelectorAll('.pipeline-show-hide .pipeline-toggle');
+        const toggle = toggles[toggles.length - 1];
+        toggle.addEventListener('click', function(event) {
+            event.preventDefault();
+            showHidePipelineSection(this);
```

Comment:
As in https://github.com/jenkinsci/job-config-history-plugin/pull/338#discussion_r1795056135, I think it would be more readable to use `toggle` instead of `button` here.